scala-package/examples/src/test/scala/org/apache/mxnetexamples/rnn/ExampleRNNSuite.scala

```diff
@@ -50,26 +49,32 @@ class ExampleRNNSuite extends FunSuite with BeforeAndAfterAll {
       System.getenv("SCALA_TEST_ON_GPU").toInt == 1) {
       ctx = Context.gpu()
     }
-    LstmBucketing.runTraining(tempDirPath + "/RNN/sherlockholmes.train.txt",
+    NDArrayCollector.auto().withScope {
+      LstmBucketing.runTraining(tempDirPath + "/RNN/sherlockholmes.train.txt",
```

Comment:
it is better to have the collector inside `runTraining`, like that you do in GanMnist.scala. Otherwise when the # of loop increases, too many ndarrays will be stored temporarily in the collector here.
But if you just mean to get rid of mem leak in CI, then this is fine.

Reply:
As you said, this piece of code is just used to improve the memory leak issues in the CI.

Reply:
@lanking520 Can you please make the change that Yizhi is asking? lets do it right because this will become a pattern in other parts of the code.

Reply:
@nswamy Depends on the purpose of the action:
1. Improving CI, this is the right action to wrap from outside
2. Improving model itself, then need to change from inside.